### ISSUE TITLE
docker containers no longer compressed

### ISSUE THREAD

**Issue by seife** [openSUSE/obs-build#451]
Kiwi creates docker containers as .tar.xz -- fine
obs-build (since commit bc95ffffdd24e1b66654d434c7b91ac61881a56b) now uncompresses them.

The comment in this commit states the compression is useless because the contents of the container are compressed anyway. This is wrong:

```$ dd if=SLES_12SP2_docker1-docker.x86_64-1.4.1-Build1.3.tar bs=128k | pigz -1 | dd of=/dev/null bs=128k
13837+1 records in
13837+1 records out
1813673721 bytes (1,8 GB, 1,7 GiB) copied, 26,2982 s, 69,0 MB/s
320+13753 records in
320+13753 records out
733005555 bytes (733 MB, 699 MiB) copied, 26,3021 s, 27
```

Clearly it can be compressed quite well.

```$ tar tvpf SLES_12SP2_docker1-docker.x86_64-1.4.1-Build1.3.tar | head
drwxr-xr-x root/root         0 2018-07-09 17:05 lib/
-rwxr-xr-x root/root     87476 2016-12-13 09:44 lib/libz.so.1.2.8
drwxr-xr-x root/root         0 2018-07-09 17:03 lib/lsb/
-rwxr-xr-x root/root       697 2017-07-07 14:12 lib/lsb/init-functions
-rwxr-xr-x root/root     25352 2018-02-26 13:24 lib/libnss_dns-2.22.so
drwxr-xr-x root/root         0 2018-07-09 17:04 lib/firmware/
-rw-r--r-- root/root      1889 2017-05-30 21:23 lib/firmware/LICENCE.i2400m
drwxr-xr-x root/root         0 2018-07-09 17:03 lib/firmware/qlogic/
-rw-r--r-- root/root     18482 2017-05-30 21:23 lib/firmware/qlogic/isp1000.bin
-rw-r--r-- root/root      8192 2017-05-30 21:23 lib/firmware/rt2661.bin
```

It contains just a plain root file system.

This commit would be good for PXE images:
```$ tar tvpf SLES_12SP3_pxeimage1.x86_64-0.6.9-Build2.116.install.tar.xz 
-rw-r--r-- root/root        58 2018-07-07 03:35 SLES_12SP3_pxeimage1.x86_64-0.6.9.md5
-rw-r--r-- root/root 373070044 2018-07-07 03:35 SLES_12SP3_pxeimage1.x86_64-0.6.9.xz
-rw-r--r-- root/root 110384368 2018-07-07 03:35 initrd-oemboot-suse-SLES12.x86_64-2.7.1.splash.install.gz
-rw-r--r-- root/root   6066000 2018-07-07 03:30 initrd-oemboot-suse-SLES12.x86_64-2.7.1.kernel.4.4.138-94.39-default
-rw-r--r-- root/root        34 2018-07-07 03:31 SLES_12SP3_pxeimage1.x86_64-0.6.9.append
```
...but there ist does not apply... :-)

**Comment by mlschroe** [comment]
containers created by dockerd contain uncompressed layers, but the containers created by kiwi with umoci/skopeo all have compressed layers. Can you point me to a container on OBS where this is not the case?

**Comment by mlschroe** [comment]
(And how can it "contain just a plain root file system"? The tar file is supposed to contain the layers and a manifest.json file, otherwise it's not a valid container.)

**Comment by seife** [comment]
This is with old kiwi in my own obs instance:

```> osc buildinfo images x86_64|grep kiwi
  <bdep name="kiwi-requires" version="7.04.40" release="45.1" arch="noarch" project="Infrastructure:KIWI" repository="SLES12" />
  <bdep name="kiwi" version="7.04.40" release="45.1" arch="x86_64" project="Infrastructure:KIWI" repository="SLES12" />
  <bdep name="kiwi-tools" version="7.04.40" release="45.1" arch="x86_64" project="Infrastructure:KIWI" repository="SLES12" />
```

This is the image type from kiwi xml:
```  <preferences>
    <type image="docker" container="sles12_sp2_bm_x86_64">
          <machine>
              <vmdisk/>
              <vmnic interface="eth0" mode="veth"/>
          </machine>
    </type>
    <version>1.4.1</version>
    <packagemanager>zypper</packagemanager>
    <rpm-check-signatures>false</rpm-check-signatures>
    <rpm-force>false</rpm-force>
    <timezone>UTC</timezone>
    <hwclock>utc</hwclock>
    <locale>en_US</locale>
  </preferences>
```

**Comment by mlschroe** [comment]
Now I'm totally confused. How can kiwi-7 build a container? I always thought that container building is a kiwi8 feature (i.e. the python one)? How does the logfile of the build look like?

**Comment by mlschroe** [comment]
"Fixed" with commit d79db29428387572ff2deff129d8825574d3baa1

**Comment by seife** [comment]
Actually my users are using "docker" image descriptions to just get a tarred rootfs, to deploy with a custom hardware-installer ;-)
They couldn't care less about containers...

This is a reproducer in build.opensuse.org (once it finishes building... :-) )
https://build.opensuse.org/package/show/home:seife:kiwitest/testimage
